### PR TITLE
Dev: Warn about adding GitHub CLI to .devcontainer

### PR DESCRIPTION
**Pull request by DLehenbauer**
Preinstall the GitHub CLI ('gh') for Codespaces and Dev Containers

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -43,9 +43,20 @@ RUN rm -rf /tmp/library-scripts/
 
 # Install additional desired packages here
 RUN export DEBIAN_FRONTEND=noninteractive \
+    && apt-get update \
```
Comment (DLehenbauer) [inline .devcontainer/Dockerfile:46]: Opportunistic improvement: Due to Docker caching, it's considered best practice to include 'apt update' before each 'apt install'.

**Review by justus-camp — DISMISSED** [review]
Nice! :shipit:

**Comment by tylerbutler** [comment]
This was removed previously; see https://github.com/microsoft/FluidFramework/pull/11587

**Comment by tylerbutler** [comment]
Assuming we need to keep it out, recommend adding a comment to the dev container file to warn folks against adding it.

**Comment by justus-camp** [comment]
Hmm, I wouldn't have thought there would be a security issue here but I'll revoke my approval regardless

**Comment by DLehenbauer** [comment]
@tylerbutler - Thank you for catching.  I'll add the warning for now and follow up.

**Review by tylerbutler — APPROVED** [review]

**Comment by DLehenbauer** [comment]
@tylerbutler - Thanks for finishing the job for me.  :-)

**Pull request merged**